src/applications/pensions/pages/spouseMarriageHistory.js

```diff
@@ -0,0 +1,111 @@
+import get from '@department-of-veterans-affairs/platform-forms-system/get';
+
+import fullSchemaPensions from 'vets-json-schema/dist/21P-527EZ-schema.json';
+
+import {
+  fullNameUI,
+  fullNameSchema,
+  currentOrPastDateUI,
+} from '@department-of-veterans-affairs/platform-forms-system/web-component-patterns';
+
+import { contactWarning, contactWarningMulti } from '../helpers';
+
+import SpouseMarriageView from '../components/SpouseMarriageView';
+import SpouseMarriageTitle from '../components/SpouseMarriageTitle';
+
+import { validateAfterMarriageDate } from '../validation';
+
+const { marriages } = fullSchemaPensions.definitions;
+
+const marriageProperties = marriages.items.properties;
+
+const reasonForSeparation = {
+  ...marriageProperties.reasonForSeparation,
+  enum: ['Death', 'Divorce', 'Other'],
+};
+
+const hasMultipleMarriages = form =>
+  get(['spouseMarriages', 'length'], form) > 1;
+
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
+    'ui:title': SpouseMarriageTitle,
+    'view:contactWarning': {
+      'ui:description': contactWarning,
+      'ui:options': {
+        hideIf: form => hasMultipleMarriages(form),
+      },
+    },
+    'view:contactWarningMulti': {
+      'ui:description': contactWarningMulti,
+      'ui:options': {
+        hideIf: form => !hasMultipleMarriages(form),
+      },
+    },
+    spouseMarriages: {
+      'ui:options': {
+        itemName: 'former marriage of the spouse',
+        viewField: SpouseMarriageView,
+        reviewTitle: 'Spouse’s former marriages',
+      },
+      items: {
+        spouseFullName: fullNameUI(title => `Former spouse’s ${title}`),
+        reasonForSeparation: {
+          'ui:title': 'How did the marriage end?',
+          'ui:widget': 'radio',
+        },
+        otherExplanation: {
+          'ui:title': 'Please specify',
+          'ui:options': {
+            hideLabelText: true,
+            expandUnder: 'reasonForSeparation',
+            expandUnderCondition: 'Other',
+          },
+          'ui:required': (form, index) =>
+            get(['spouseMarriages', index, 'reasonForSeparation'], form) ===
+            'Other',
+        },
+        dateOfMarriage: currentOrPastDateUI('Date of marriage'),
+        dateOfSeparation: {
+          ...currentOrPastDateUI('Date marriage ended'),
+          'ui:validations': [validateAfterMarriageDate],
```

Comment:
While manually testing I noticed that the `validateAfterMarriageDate` validation doesn't appear to be working correctly. I was able to save a marriage end date before the marriage start date.
![Screenshot 2023-12-08 at 8 24 29 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/a433a1db-023b-4d9f-b9b1-934a7f02cd95)